### PR TITLE
add option n for update command to not save config

### PR DESCRIPTION
**Pull request by thojansen**
While trying to set up wifi manually via ssh and uci commands, my wifi settings were somehow broken. Even manually setting the wifi settings to another ad-hoc network was no longer working as only parts are overwritten by the cli tool and ssh was no longer working as there was no wifi possible anymore. On the other hand the update did not reset my configuration, so I essentially needed this fix to unbrick my tessel2. Any feedback is appreciated to get this merged. Workaround was originally posted in #619. @johnnyman727

### CODE REVIEW

File: lib/controller.js [hunk 1]
```diff
@@ -800,7 +800,7 @@ controller.printAvailableUpdates = function() {
   });
 };
 
-controller.update = function(opts) {
+controller.update = function(opts) {
```
Comment (rwaldron) [inline lib/controller.js]: This shouldn't be here ;)

You need to run `grunt` locally, which will clean up things like that for you and also ensures that all the linting and tests are run. Ping me after you've had a chance to do that.

File: bin/tessel-2.js [hunk 2]
```diff
@@ -353,6 +353,12 @@ makeCommand('update')
     flag: true,
     help: 'Update to the latest version regardless of current version.'
   })
+  .option('delete', {
+    abbr: 'd',
+    required: false,
+    flag: true,
+    help: 'Delete configuration files during update.'
+  })
```
Comment (rwaldron) [inline bin/tessel-2.js]: This is a strange name for a thing associated with `t2 update`

Comment (thojansen) [inline bin/tessel-2.js]: Yepp, agreed. Here the docs.. maybe just re-use it?
https://wiki.openwrt.org/doc/techref/sysupgrade

` -n do not save configuration over reflash`

**Comment by rwaldron** [comment]
Travis is failing because this changeset breaks the update tests: https://travis-ci.org/tessel/t2-cli/builds/123382880

Please have a look and update them appropriately. You'll also need to add new tests for the new capability. Ping me here or on Slack for any help getting started with testing.

**Comment by thojansen** [comment]
@rwaldron now the option is renamed from `d` to `n` matching sysupgrade. Linting and unit tests are passing again and two additional tests were added (although i am not 100% happy with the tests).

File: test/unit/update.js [hunk 3]
```diff
@@ -350,7 +350,55 @@ exports['controller.update'] = {
...
         test.equal(this.update.callCount, 1);
         // It was provided the binaries
-        test.equal(this.update.calledWith(binaries), true);
+        test.equal(this.update.calledWith(opts, binaries), true);
+        // Then Tessel was closed
+        test.equal(this.tessel.closed, true);
+        // We closed all open Tessel connections
+        test.equal(this.closeTesselConnections.callCount, 1);
+        // We called the close function with an array
+        test.equal(Array.isArray(this.closeTesselConnections.args[0]), true);
```
Comment (rwaldron) [inline test/unit/update.js]: Great start here! For this test, use test.deepEqual to compare arg[0] with the array you expect it to be

File: test/unit/update.js [hunk 4]
```diff
@@ -679,6 +727,34 @@ exports['Tessel.update'] = {
...
+    this.sysupgradeNoSaveConfig = this.sandbox.stub(commands, 'sysupgradeNoSaveConfig');
+
+    this.tessel.updateOpenWRT({
+      n: true
+    }, this.newImage.openwrt).then(() => {
+      test.equal(this.openStdinToFile.callCount, 1);
+      test.equal(this.sysupgradeNoSaveConfig.callCount, 1);
+      test.equal(this.openStdinToFile.lastCall.args[0], updatePath);
+      test.equal(this.sysupgradeNoSaveConfig.lastCall.args[0], updatePath);
+      test.done();
```
Comment (rwaldron) [inline test/unit/update.js]: Similar to the test I described above, add a test that asserts the right args were used

**Comment by rwaldron** [comment]
This is fantastic work! I left a couple test related comments, once those are addressed, I will do a smoke test with an actual T2

**Comment by thojansen** [comment]
Added deepEqual for the first test. For the second one, the options are actually not passed to the commands, but a different command is used. I choose to implement it this way, as there was basically no logic in `commands.js`.

**Comment by rwaldron** [comment]
If you haven't done so already, please sign the CLA: http://dojofoundation.org/about/claForm

File: test/unit/update.js (hunk 3 above)
Comment (rwaldron) [inline test/unit/update.js]: Derp, this is not what I thought it was... Ignore me here ;)

**Comment by thojansen** [comment]
CLA signed

**Comment by rwaldron** [comment]
Thank you kindly :)

**Pull request merged**